### DISCUSSION TITLE
Add unfillable method to be able to create "ui fields"

### DISCUSSION THREAD

**Discussion by RomainMazB** [laravel/nova-issues#6328] [Ideas & Feature Requests] [answered]
As of today, some Field already implements the `Unfillable` interface, like the `Header` to create a Vue component which doesn't hold any value that won't fill the model before saving it to the database.

The feature request is to expand this feature to field using an `unfillable` method instead of the interface.

## Why this is needed
A basic use case is to be able to use a select dropdown where the value is looked from other fields using `dependsOn` feature to hide/show some fields.

I have a Loot model, where the amount is generated using 3 fields: `amount_fixed`, `amount_min`, `amount_max`.
`amount_fixed` is only used when I want the loot to be "fixed": always 3.
`amount_min` and `amount_max` are used when I want the loot amount to be random: between 1 and 5.
![fixed-or-random](https://github.com/laravel/nova-issues/assets/53976837/f1434cff-1969-41f7-a998-78f80798693f)

```php
// This field doesn't exist in database, it's just live for UX improvement
Select::make('Amount mode', 'amount_mode')
    ->onlyOnForms()
    ->fillUsing(fn () => null)
    ->options([
        'random' => 'Random',
        'fixed' => 'Fixed',
    ])
    ->default(fn(NovaRequest $request) => is_null($request->resource->amount_fixed) ? 'random' : 'fixed'),

// This field is only displayed when the dropdown value is "fixed"
Number::make('Amount', 'amount_fixed')
    ->dependsOn(
        'quantities',
        fn (Number $field, NovaRequest $request, FormData $formData) =>
            $formData->quantities === 'fixed' ? $field->show() : null
    ),


// These fields are only displayed when the dropdown value is "random"
Number::make('Amount min')
    ->dependsOn(
        'quantities',
        fn (Number $field, NovaRequest $request, FormData $formData) =>
            $formData->quantities === 'random' ? $field->show() : null
    ),

Number::make('Amount max')
    ->dependsOn(
        'quantities',
        fn (Number $field, NovaRequest $request, FormData $formData) =>
            $formData->quantities === 'random' ? $field->show() : null
    )
```
All these three fields are nullable and I have a dropdown to select whether I want the Loot to be fixed or random.

Today, to not persist the amount mode value to the database using the `->fillUsing(fn () => null)` and `->onlyOnForms()` because there isn't no value to display on index/preview but it would be clearer if we could easily apply this trick using an `->unfillable()` method.

**Comment by crynobone** [comment, marked as answer]
Released with Laravel Nova 5.0.0

```diff
Select::make('Amount mode', 'amount_mode')
-   ->onlyOnForms()
-   ->fillUsing(fn () => null)
+   ->computed()
```

Feel free to open up a new issue if you're still experiencing this problem on the latest version.
